[Lee Rong Jieh] iP

lrj689 opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

lrj689 committed:
Update Duke

Added increment Level-1: Greet, Echo, Exit

lrj689 committed:
Update Duke Level-1

tidy up code

no functional changes

lrj689 committed:
Added DukeList.java

Updated Duke.java

DukeList class functions as a custom ArrayList to store user inputs

Duke class updated to use DukeList class for user inputs and outputs

lrj689 committed:
Update DukeList.java with print method

Print method prints DukeList object in Duke.java

lrj689 committed:
Added Task.java

Updated DukeList.java, Duke.java

Task is a class representing a String and a boolean

Updated DukeList use Task objects

lrj689 committed:
Add Deadlines.java, ToDos.java, Events.java

DukeList class: updated functionality to use new classes as mentioned above

Duke class: cleaned up text output

lrj689 committed:
Update runtest.bat

lrj689 committed:
Add DukeExceptions.java

Update Duke to handle exceptions

lrj689 committed:
Update Duke.java, DukeList.java

DukeList: add delete() method

Duke: updated to use delete method from DukeList and to handle new exceptions introduced

lrj689 committed:
Update Duke.java to save changes

lrj689 committed:
Add LocalDate

lrj689 committed:
Duke class: update to use Date format, add show command

show command: takes in a date and prints all task on that given date

banchiang commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadlines extends Task {
+
+    private String by;
```
Similiar mistake as me! We can change the variable name 'by' to be more intuitive

SoulUseless replied:
Since this class indicates a singular Deadline Task object, it could be better if you named your class in the singular form instead of the plural form. This also reduces confusion should the need for having a Deadline array arise.

banchiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,115 @@
+
+        while (sc.hasNextLine()) {
+            String input = sc.nextLine();
+            String[] arr = input.split("/");
```
String array name can be more descriptive :D

YuFeng0930 replied:
Agree with you, since the scope of arr is quite big.

banchiang commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,115 @@
+            } else {
+                // unknown command
+                System.out.println(topBound + "OOPS!!! I'm sorry, but I don't know what that means :-(" + bottomBound);
+            }
```
Since there are many if else statements, I think using switch would make the code look cleaner :)

banchiang commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+     */
+
+    private String taskName;
+    private boolean done;
```
As per coding standard, boolean variable name would be a name to point out that the variable is boolean. E.g ("isDone") would be good :D

banchiang reviewed (commented):
Review for ip for coding standards. Do have a look :D

SoulUseless commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,115 @@
+                    list.done(x);
+                    System.out.println(topBound + "Good job! I've marked this task as done:\n" + "  " + list.get(x)
+                            + bottomBound);
+                } catch (IndexOutOfBoundsException e) {
+                    // Task number x does not exist
+                    System.out.println(topBound + "Task does not exist and can not be completed!" + bottomBound);
+                } catch (NumberFormatException e) {
+                    // parameter for done is not an Integer
+                    System.out.println(topBound + "done must be followed by an Integer ie. done 1. Try again!"
+                        + bottomBound);
+                }
```
I have noticed that you have implemented DukeException.java as a error handling class. Perhaps you can leverage that to handle your exceptions instead of printing them here?

SoulUseless commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,115 @@
+            }
+        }
     }
 }
```
I like how thorough your error handling is. However, the main function is very long. Would it be better to abstract some if-block code into separate functions to aid readability?

SoulUseless commented on src/main/java/DukeList.java:
```diff
@@ -0,0 +1,50 @@
+        list.add(item);
+    }
+
+    public void done(int x) {
```
Since this is a function that does an action, would it be better to name it in a way that it sounds like a function?

SoulUseless commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,15 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " + duration + ")";
```
I like your use of the inherited toString() method.

SoulUseless reviewed (commented):
Please find below my PR review.

lrj689 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadlines.java
#	src/main/java/Duke.java
#	src/main/java/DukeList.java
#	src/main/java/Events.java
#	src/main/java/Task.java

lrj689 committed:
Duke class: Add reset command, cleaned up code

reset command: deletes all current tasks

YuFeng0930 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,190 @@
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String bound = "____________________________________________________________";
```
Constant names must be all uppercase.

YuFeng0930 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,190 @@
+            }
+        }
+    }
+    public static void readFromFile(String filePath, DukeList list) throws FileNotFoundException {
```
There should be a line break to separate functions. JavaDoc for public functions need to be added also.

YuFeng0930 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,190 @@
+import java.util.Scanner;
+
 public class Duke {
     public static void main(String[] args) {
```
Cannot write everything in main function. Even if it's not main function, this function itself is too long also, which reduce the readability of the code.

YuFeng0930 commented on src/main/java/DukeList.java:
```diff
@@ -0,0 +1,103 @@
+     * @param x Task number to be removed
+     * deletes the stated task
+     */
+
```
There shouldn't be a line break between JavaDoc and function header.

YuFeng0930 reviewed (commented):
The first thing needs to be done is move things out of main function. Try to break each functions into more functions, which will make the code much understandable and easier to maintain.

lrj689 committed:
Add Parser.java UI.java

lrj689 committed:
Update UI.java

lrj689 committed:
Add Storage.java
Updated Duke.java, DukeList.java, Parser.java, UI.java
Storage: Store and read the contents of DukeList
Others: Update functionality

lrj689 committed:
Add JUI tests

lrj689 committed:
no message

lrj689 committed:
Add JavaDocs

lrj689 committed:
Adhere to java coding standard

lrj689 committed:
Update Find functionality
Find: outputs tasks that matches keyword provided

lrj689 committed:
Update Ui.java, Parser.java
Update to use String[] as arg for more flexibility

lrj689 committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/Parser.java
#	src/main/java/UI.java

lrj689 committed:
Merge branch 'branch-A-CodingStandard'

lrj689 committed:
Merge branch 'branch-level-9'

# Conflicts:
#	src/main/java/Parser.java
#	src/main/java/UI.java

lrj689 committed:
Update UI.java, Parser.java

Improved text output quality

lrj689 committed:
Update Duke.java, DukeList.java, Storage.java
Cleaned up code

lrj689 committed:
Update Storage.java, Task.java
Clean up code

lrj689 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

lrj689 committed:
Add Gradle Support
Update DukeListTest.java

lrj689 committed:
Add Main.java
Add MainWindow.java
Add Laucher.java
Add DialogBox.java
Add DialogBox.fxml
Add MainWindow.fxml

lrj689 committed:
Add Laucher.java
Add Main.java
Add MainWindow.java
Add DialogBox.java
Add DialogBox.fxml
Add MainWindow..fxml

Skeletal GUI implemented

lrj689 committed:
Update Duke.java
Update UI.java
Duke : updated to launch GUI
UI: updated to return Strings

lrj689 committed:
Update GUI with exit functionality

lrj689 committed:
Add Package dukePackage
Update userImage
Update dukeImage

lrj689 committed:
Update package name

lrj689 committed:
Add Assertions

lrj689 committed:
Update Storage.java
Clean up code

lrj689 committed:
Add more Assertions

lrj689 committed:
Add Assertions

lrj689 committed:
Update DialogBox.java
Update DialogBox.fxml

Fixed issue when long text is overriden by the next text

lrj689 committed:
Add CheckStyle

lrj689 committed:
Update package name

lrj689 committed:
Improved Code Quality

Removed DukeException due to redundancy

lrj689 committed:
Delete ParserTest.java
Update DukeList.java
Update DukeListTest.java

ParserTest.java not used hence deleted

DukeList deleteAll() method updated to use clear() method from ArrayList

More tests added to DukeListTest

lrj689 committed:
Add help command

help command displays all usable commands to the user

lrj689 committed:
Update duke.txt

lrj689 committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	data/duke.txt

lrj689 committed:
Merge pull request #3 from lrj689/branch-A-CheckStyle

Add CheckStyle

lrj689 committed:
Merge pull request #4 from lrj689/branch-A-CodeQuality

Improved Code Quality

lrj689 committed:
Update Welcome and Goodbye messages

lrj689 committed:
Update UI.java

lrj689 committed:
Merge branch 'master' into branch-C-Help

lrj689 committed:
Merge pull request #5 from lrj689/branch-C-Help

Branch c help

lrj689 committed:
Update GUI

lrj689 committed:
Update help command

Improved help text clarity

lrj689 committed:
Update UserGuide

lrj689 committed:
Update README.md

lrj689 committed:
README.md: fixed some formating issues

lrj689 committed:
Rename README.md to doc/README.md

lrj689 committed:
Rename docs/README.md to README.md

lrj689 committed:
Fix file structure

lrj689 committed:
Set theme jekyll-theme-slate

lrj689 committed:
Update README.md
